lunes_cms/help/templates/public_upload.html

```diff
@@ -65,28 +65,46 @@
         3: "das",
         4: "die (Plural)"
       }
+      var buffer_array = []
 
-      $( document ).ready(function() {
+      function Comparator(a, b) {
+        if (a[1] < b[1]) return -1;
+        if (a[1] > b[1]) return 1;
+        return 0;
+      }
 
+      $( document ).ready(function() {
+        $("#inputDiscipline").append("<option value='0'>Alle Berufe</option>")
+        lunes_disciplines = lunes_disciplines.sort(Comparator);
         lunes_disciplines.forEach(function(element) {
           if( element[1] !== "Allgemeine Grundlagen") {
             $("#inputDiscipline").append(new Option(element[1], element[0]))
           }
         });
-
+        
         $("#inputDiscipline").change(function() {
           var discipline_id = $("#inputDiscipline").val();
-          $("#inputDocument").find('option').remove();
-          lunes_training_sets.forEach(function(element) {
-            if(lunes_disc_sets_map[discipline_id].includes(element[0])) {
-              var training_set_id = element[0];
-              lunes_documents.forEach(function(element) {
-                if(element[3] == training_set_id) {
-                  $("#inputDocument").append(new Option( article_map[element[2]] + " " + element[1], element[0]));
-                }
-              });
-            }
-          });
+          if (discipline_id !== "0") {
+            $("#inputDocument").find('option').remove();
+            lunes_training_sets.forEach(function(element) {
+              if(lunes_disc_sets_map[discipline_id].includes(element[0])) {
+                var training_set_id = element[0];
+                lunes_documents.forEach(function(element) {
+                  if(element[3] == training_set_id) {
+                    $("#inputDocument").append(new Option( article_map[element[2]] + " " + element[1], element[0]));
+                  }
+                });
+              }
+            });
+          } else {
+            options = []
+            lunes_documents.forEach(function(element) {
+              options.push(new Option( article_map[element[2]] + " " + element[1], element[0]))
+            })
+            options.forEach(function(element){
+              $("#inputDocument").append(element)
+            })
+          }
```

Comment:
I think the current code is a lot more complicated than needed. Instead of trying to wrap your head around this complex nested loops and try to find a way to sort the documents, I would suggest to refactor the whole data structure how the disciplines and documents are passed to JS here.
Would it be ok for you if I refactored the whole public upload page and took over the sorting issue?